_episodes/03-monitor-the-model.md

```diff
@@ -449,110 +454,110 @@ history = model.fit(X_train, y_train,
-> * How low can you go with the number of parameters without notable effect on the performance on the validation set?
->
-> > ## Solution
-> > ~~~
-> > def create_nn(nodes1, nodes2):
-> >     # Input layer
-> >     inputs = keras.layers.Input(shape=(X_data.shape[1],), name='input')
-> >
-> >     # Dense layers
-> >     layers_dense = keras.layers.Dense(nodes1, 'relu')(inputs)
-> >     layers_dense = keras.layers.Dense(nodes2, 'relu')(layers_dense)
-> >
-> >     # Output layer
-> >     outputs = keras.layers.Dense(1)(layers_dense)
-> >
-> >     return keras.Model(inputs=inputs, outputs=outputs, name="model_small")
-> >
-> > model = create_nn(10, 5)
-> > model.summary()
-> > ~~~
-> > {:.language-python}
-> >
-> > ~~~
-> > Model: "model_small"
-> > _________________________________________________________________
-> > Layer (type)                 Output Shape              Param #   
-> > =================================================================
-> > input (InputLayer)           [(None, 163)]             0         
-> > _________________________________________________________________
-> > dense_21 (Dense)             (None, 10)                1640      
-> > _________________________________________________________________
-> > dense_22 (Dense)             (None, 5)                 55        
-> > _________________________________________________________________
-> > dense_23 (Dense)             (None, 1)                 6         
-> > =================================================================
-> > Total params: 1,701
-> > Trainable params: 1,701
-> > Non-trainable params: 0
-> > _________________________________________________________________
-> > ~~~
-> > {:.output}
-> >
-> > ~~~
-> > model.compile(optimizer='adam',
-> >               loss='mse',
-> >               metrics=[keras.metrics.RootMeanSquaredError()])
-> > history = model.fit(X_train, y_train,
-> >                     batch_size = 32,
-> >                     epochs = 200,
-> >                     validation_data=(X_val, y_val), verbose = 2)
-> >                     
-> > history_df = pd.DataFrame.from_dict(history.history)
-> > sns.lineplot(data=history_df[['root_mean_squared_error', 'val_root_mean_squared_error']])
-> > plt.xlabel("epochs")
-> > plt.ylabel("RMSE")
-> > ~~~
-> > {:.language-python}
-> >
-> > ![Output of plotting sample](../fig/03_training_history_3_rmse_smaller_model.png)
-> >
-> > There is obviously no single correct solution here. But you will have noticed that the number of nodes can be reduced quiet a bit!
-> >
-> > In general, it quickly becomes a very complicated search for the right "sweet spot", i.e. the settings for which overfitting will be (nearly) avoided but which still performes equally well.
-> >
-> {:.solution}
-{:.challenge}
+We can keep the network architecture unchanged (2 dense layers + a one-node output layer) and only play with the number of nodes per layer.
```

Comment:
I think that we can actually keep this exercise as long as we don't use the parametrized `create_nn` function in the rest of the main lesson material.

In this exercise the participants are asked to play with the number of nodes but in the rest of the lesson content we just continue with the initially defined number of nodes.

Reply:
good point. Do I understand correctly, that you'd like to see the exercise being inserted back or in a more condensed fashion perhaps?

Reply:
Yes exactly

Reply:
Let's handle this point in a different PR. I'm already working on it because I am preparing for a workshop tomorrow.
We can remove the exercises in this PR and I'll make a different PR where I add back in some selected exercises (or parts of them)